src/errorReporter.ts

```diff
@@ -0,0 +1,68 @@
+
+import * as vscode from 'vscode';
+
+export const enum ErrorCodes {
+  Error,
+  Warning,
+  Informational
+}
+
+export interface ErrorEventInterface {
+  title: string,
+  message: string | unknown,
+  errorCode: ErrorCodes
+}
+
+export class ErrorReporter {
+  static async showErrorDialog(
+    error: ErrorEventInterface): Promise<void> {
+
+    const template = `<!-- Please delete any private information -->
+        **Version:**
+        ${// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+          vscode.extensions.getExtension('ms-edgedevtools.vscode-edge-devtools')?.packageJSON.version || 'N/A'}
+
+        **OS:**
+        <!-- Add your hosting platform e.g Microsoft Windows, WSL -->
+
+        **Stack :**
+        ${error.message}
+
+        **Additional context:**
+        <!-- Add any other context or about the problem or screenshots here. -->`;
+
+    // cannot do multiline due to:
+    // https://github.com/Microsoft/vscode/issues/48900
+    const answer = await vscode.window
+      .showErrorMessage(
+        `An error occurred:\n${error.title}\n${error.message}`,
```

Comment:
nit: should we even include the '\n' since it doesn't do anything?

Reply:
Thanks for the review @bgoddar , regarding the telemetry I made the same question myself at some point I even thought one was a superset of the other but eventually got to a similar conclusion; there a couple of high priority scenarios where is better to have them split, so most likely they just have an intersection, so it is better to keep them separate (at least ATM)